src/local_scheduler/local_scheduler.cc

```diff
@@ -148,6 +149,11 @@ void LocalSchedulerState_free(LocalSchedulerState *state) {
    * local scheduler at most once. If a SIGTERM is caught afterwards, there is
    * the possibility of orphan worker processes. */
   signal(SIGTERM, SIG_DFL);
+  /* Send a null heartbeat that tells the global scheduler that we are dead to
+   * avoid waiting for the heartbeat timeout. */
+  if (state->db != NULL) {
+    local_scheduler_table_disconnect(state->db);
+  }
```

Comment:
Yes, that's true. When I wrote the `local_scheduler_table_disconnect` call, I tried to match it to the current codepath as much as possible, i.e. the global scheduler sets the last heartbeat to long ago instead of deleting the local scheduler immediately. Therefore, it *should* be as if we just sped up the time to timeout the local scheduler, but it's hard to say if we covered all corner cases.

Probably the resolution to this is to do some more thorough testing with a live cluster, where we abruptly kill nodes. This is on the backlog.

Reply:
Is the only point of this to speed up the tests? If so, another alternative is to just use a smaller number of missed heartbeats to trigger cleanup in the tests. If you think that makes sense we can do it in a separate PR.

Reply:
It's also for cases where the local scheduler doesn't exit fatally. I think it's useful to keep it here so that we can do things like scale down a live cluster without having to wait for the full heartbeat timeout.